### PR TITLE
e4s cray sles ci: expand spec list

### PR DESCRIPTION
**Pull request by eugeneswalker**

### CODE REVIEW

File: share/spack/gitlab/cloud_pipelines/stacks/e4s-cray-rhel/spack.yaml [hunk 1]
```diff
@@ -21,47 +21,156 @@ spack:
...
+    trilinos:
+      require:
+      - one_of: [+amesos +amesos2 +anasazi +aztec +boost +epetra +epetraext +ifpack
+            +intrepid +intrepid2 +isorropia +kokkos +minitensor +nox +piro +phalanx
+            +rol +rythmos +sacado +stk +shards +stratimikos +tempus +tpetra
+            +trilinoscouplings +zoltan]
+      - one_of: [gotype=long_long, gotype=all]
+      - one_of: [~ml ~muelu ~zoltan2 ~teko, +ml +muelu +zoltan2 +teko]
+      - one_of: [+superlu-dist, ~superlu-dist]
+      - one_of: [+shylu, ~shylu]
```
Comment (alalazo) [inline share/spack/gitlab/cloud_pipelines/stacks/e4s-cray-rhel/spack.yaml]: I was looking at other E4S stacks a couple of days ago, and noticed the same. Aren't these two lines redundant (requiring a Boolean variant to be either true or false)? What was their purpose when they were added?

Comment (eugeneswalker) [reply share/spack/gitlab/cloud_pipelines/stacks/e4s-cray-rhel/spack.yaml]: I don't recall. There was a concretization issue and this actually helped resolve it at one point. All of the `one_of` were added together to resolve the issue in the past. Perhaps it is not needed anymore? If you say these two lines have no effect, let us remove them and see what happens?

**Comment by eugeneswalker** [comment]
@spackbot run pipeline

**Comment by eugeneswalker** [comment]
All CI is green! I think we are ready for this.

**Review by wspear — APPROVED** [review]

**Pull request merged**